kano_updater/scenarios.py

```diff
@@ -860,4 +860,18 @@ def beta_3_11_0_to_beta_3_12_0(self):
         remove_user_files(['.asoundrc'])
 
     def beta_3_12_0_to_beta_3_12_1(self):
-        pass
+        try:
+            '''
+            Any CKC user passing through this function must have updated and
+            hence must be a v1.0 user (CKC v1.1 ships with a higher version
+            number and has logic to determine the version by the hardware).
+
+            Add flag for these users to show a free speaker upgrade message.
+            '''
+
+            from kano_peripherals.wrappers.detection import is_ck2_pro
+            if is_ck2_pro():
+                speaker_warning_file = '~/show_speaker_warning'
```

Comment:
Mmmmm... I think this will point to `/root` as the updater is sudoed?
You have `SUDO_USER` but I'm sure we address this case on other places.

Reply:
I'm so silly... the function does it for us, sorry.

Reply:
@skarbat We run this through a `run_for_every_user()` so we should be okay.